producer.go

```diff
@@ -279,6 +285,38 @@ func (opts *ProduceOpts) produce(p *Producer) error {
 	}
 }
 
+func (p *Producer) validateMsg(msg any) ([]byte, error) {
+	sd, err := p.getSchemaDetails()
+	if err != nil {
+		return nil, err
+	}
+
+	if sd.schemaType == "" {
+		switch msg.(type) {
+		case []byte:
+			return msg.([]byte), nil
+		default:
+			return nil, errors.New("Unsupported message type")
+		}
+
+	}
+
+	return sd.validateProtoMsg(msg)
+}
+
+func (p *Producer) getSchemaDetails() (schemaDetails, error) {
+	return p.conn.getSchemaDetails(p.stationName)
```

Comment:
please make sure that your map keys are always internally names here and everywhere else